### PR TITLE
[Train][Workspace template] Updated the workspace template with new fixes

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Why are these changes needed?

**1. Updated the Readme:**
* Added time to run fields for all models (after running all variations at least once)
* Fix CLI typo
* Added guideline on how to pick the instance types
 
**2. Updated the cluster config**
* Base image to 2.6.1 and cuda117
* use pip package list instead of post build commands
* added debian packages for enabling nvme offloading

**3. Added and modified compute configs**
* Added two versions of compute config for 70B
* Modified the other compute configs to include custom resourse

**4. Updated Deepspeed configs**

**5. Modified the fine-tuning script**
* padding=`longest` --> padding=`max_length`: This will ensure we are measuring the perf. consistently with a fixed block size
* Evaluation method had some bugs around aggregation that are fixed 
* Added `ACCELERATE_TORCH_DEVICE` setup directly into the training function until this is fixed by ray train
* Moved the tmp directory for checkpoint from root to output dir which allows it to be on nvme: This opens up capacity for more disk when checkpointing a model as big as 70B.
* Enabled sharding for evaluation dataset which speeds up evaluation by 16x for 7/13B and 32x for 70B.

**6. Fix the `--as-test` flag bugs in the shell scripts**

**7. Added nightly release tests for 7B and 13B**

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kouroshHakha** [comment]
release tests: https://buildkite.com/ray-project/release-tests-pr/builds/46618

**Comment by kouroshHakha** [comment]
release-test: https://buildkite.com/ray-project/release-tests-pr/builds/46647#01899314-eefa-42cb-a658-33437d56639d

File: doc/source/templates/04_finetuning_llms_with_deepspeed/cluster_env.yaml [hunk 1]
```diff
@@ -1,31 +1,33 @@
 # See https://hub.docker.com/r/anyscale/ray for full list of
 # available Ray, Python, and CUDA versions.
-base_image: anyscale/ray:2.5.1-py39-cu116
+base_image: {{ env["RAY_IMAGE_ML_NIGHTLY_GPU"] | default("anyscale/ray:2.6.1-py39-cu117") }}
```
Comment (aslonnie) [inline doc/source/templates/04_finetuning_llms_with_deepspeed/cluster_env.yaml:3]: why use nightly? why not use a released one?

also why not use cu118?

File: doc/source/templates/04_finetuning_llms_with_deepspeed/run_llama_13b.sh [hunk 2]
```diff
@@ -23,30 +23,10 @@ if [ ! -d "${DATA_DIR}" ]; then
...
-
-# Construct additional arguments string
-additional_args=""
-for key in "${!args[@]}"; do
-    additional_args+="${args[$key]} "
-done
+# Check if --as-test was passed as an argument
+if [[ "$@" == *"--as-test"* ]]; then
+    params+=" --as-test"
+fi
```
Comment (aslonnie) [inline doc/source/templates/04_finetuning_llms_with_deepspeed/run_llama_13b.sh:29]: why not keep using the for loop? which is a much better check?

File: doc/source/templates/04_finetuning_llms_with_deepspeed/run_llama_13b.sh [hunk 3]
```diff
@@ -59,6 +39,6 @@ python finetune_hf_llm.py \
     --train_path ${TRAIN_PATH} \
     --test_path ${TEST_PATH}  \
     --special_token_path ${TOKEN_PATH} \
-    "$additional_args"
+    $params
```
Comment (aslonnie) [inline doc/source/templates/04_finetuning_llms_with_deepspeed/run_llama_13b.sh:42]: use a bash array and here `"${params}[@]"`

also in the other ones.

File: release/ray_release/buildkite/concurrency.py [hunk 4]
```diff
@@ -16,7 +16,7 @@
 )
 
 aws_gpu_cpu_to_concurrency_groups = [
-    Condition(min_gpu=9, max_gpu=-1, min_cpu=0, max_cpu=-1, group="large-gpu", limit=4),
+    Condition(min_gpu=9, max_gpu=-1, min_cpu=0, max_cpu=-1, group="large-gpu", limit=100),
```
Comment (aslonnie) [inline release/ray_release/buildkite/concurrency.py:19]: why is this change required?

how many do you really need? 4 to 100 is a pretty big jump.

**Review by aslonnie — COMMENTED** [review]
this PR seems to be packing many changes into one. should we split it into smaller ones?

File: doc/source/templates/04_finetuning_llms_with_deepspeed/cluster_env.yaml (hunk 1 above)
Comment (kouroshHakha) [reply doc/source/templates/04_finetuning_llms_with_deepspeed/cluster_env.yaml:3]: to always test this with the latest on CI?

**Comment by kouroshHakha** [comment]
@aslonnie Didn't mean to ping ray-ci folks. @can-anyscale suggested changing the concurrency temporarily so I can get through checking the release tests. This is a new feature and won't impact anything else. I rather pack all the changes I can in one PR in favor of speed here.

**Comment by kouroshHakha** [comment]
Bypassing release tests for now so that we can proceed with dogfooding.

**Comment by aslonnie** [comment]
>  I can in one PR in favor of speed here.

I disagree. it will make delivery slower when something is broken by the PR and needs to be reverted, also makes reviewing harder.

but anyways.. seems that you do not want my review anyways, so I will just fade.. :)
